1.1/spec/09-Part-06.adoc

```diff
@@ -32,6 +32,25 @@ geo:SpatialObject a rdfs:Class,
 |`/req/core/spatial-object-class`
 |===
 
+===== Class: geo:SpatialMeasure
+
+The class `geo:SpatialMeasure` is defined by the following:
+
+```
+geo:SpatialMeasure a rdfs:Class, 
+                    owl:Class;
+    rdfs:isDefinedBy <http://www.opengis.net/spec/geosparql/1.1>;
+    rdfs:label "Spatial Measure"@en;
+    rdfs:comment "The class represents a measurement of a feature bot only 
+                spatial measurements. It is therefore naturally a subclass 
+                of Spatial Object which is any spatial thing."@en .
```

Comment:
Not sure I agree with that. 'Measure' is from ISO 19103, and I think is a datatype, not a class. It has to be implemented in RDFS/OWL as a class because it has more than one component, but functionally it is a datatype for a scaled number.

Reply:
So what do you suggest? It is not then to be a part of the `Geometry` / `SpatialObject` hierarchy?

Reply:
Correct. It does not belong in that subsumption chain.

Reply:
OK, then how do we align the `hasArea` etc. with `hasGeometry`? Clearly `Feature hasGeometry Geometry` is similar to `Feature hasArea/hasLength/hasVolume SpatialMeasure` so what is the superclass of `Geometry` & `SpatialMeasure`? Is it to be left unstated in GeoSPARQL or is `SpatialObject` the superclass?

I'm worried about not tying all the property things together for surely soon we will have `Feature hasTime TemorpalObject` and we will have this all over again.

Should we bite the bullet and have, within GeoSPARQL, 

```
                     Measure
              ^                      ^
SpatialMeasure TemporalMeasure
     ^
Geometry
```

or similar? Or do we cut off GeoSPARQL at the Spatial? I'd prefer to include Temporal as I kinda need spatio-temporal objects.

Reply:
hasArea / hasLength etc is kinda like pointing to one component of a coordinate. It is an incomplete description. Useful, but incomplete. You might make both `hasGeometry` and `hasLength` sub-properties of `hasSpatialProperty` but IMO it isn't worth the effort. 

As I mentioned, ISO 19105 does not think Measure has a super-class

2005 version: 
![image](https://user-images.githubusercontent.com/608303/103599144-87dc2500-4f58-11eb-9a83-f859dcdb110d.png)

2013 revision: 
![image](https://user-images.githubusercontent.com/608303/103599201-b3f7a600-4f58-11eb-91a4-1f7d59bc99cd.png)

Reply:
From issue #33: 
> it is required to keep the Geometry / SpatialObject relations unchanged. 

Do I understand correctly that this is the reason for introducing the `SpatialMeasure` class? Understandable, but this introduces a deviation from ISO 19107. That may not be a good idea. 

That said, from my limited understanding of ISO 19107 (not having a background in geodesy), a geometry is a measure, although the term Measure seems to be understood (in ISO 19105) to mean other (simpler) types of quantities.

Reply:
small note: during a design process (e.g. road design, building design, etc.) geometry can be designed instead of measured.

Reply:
> You might make both hasGeometry and hasLength sub-properties of hasSpatialProperty but IMO it isn't worth the effort.

I think it might be. I do think that a geometry is just a particular kind of measurement. It may be a compound one (lots of individual lengths/points etc.) but it's still just a measure (of spatiality) just as hasTime would be a measure of temporality.

I'm keen to see geometry brought into the measurement fold.

> geometry can be designed instead of measured

I don't think this is really a real difference! It's still a property of something (a Feature) as seen from GeoSPARQL's point of view.

Reply:
I think it's just a matter of wording: I agree that geometry is a special type of property of a FoI, but I would not call it a measurement. Are the `Measure` and `TemporalMeasure` classes already defined in some ontology? BTW: instead of directly discussing TBox, we can maybe start the discussion from data level (ABox)? In the OMG/FOG/GOM approach we connect geometry and related properties (incl. spatial ones such as length and area) as follows:
```
inst:road1 omg:hasGeometry inst:road1-geom1 ;
   ex:length inst:road1-prop1 .
inst:road1-geom1 a omg:Geometry ;
   fog:asObj_v3.0-obj "an OBJ string"^^xsd:string .
inst:road1-prop1 a opm:Property , seas:Property ;
   schema:value "100.4"^^xsd:decimal ;
   qudt:unit unit:M ;
   omg:isImplicitlyDerivedFrom inst:road1-geom1 .
```
The (currently fictive) spatial property `ex:length` is not a sibling of `omg:hasGeometry` as I don't see much advantages except for more clarity on an abstract level (the same goes for the `omg:Geometry` and `opm:Property` classes: the first is no subclass of the latter). In that sense, I agree with @dr-shorthair. Personally, I think that ABox links between the property and the geometry are more interesting, e.g. to indicate that the length property is measured on (derived from) the geometry object.

Reply:
> Are the Measure and TemporalMeasure classes already defined in some ontology?

We are exploring that here. `SpatialMeasure` in our extension ontology for GeosPARQL (https://linked.data.gov.au/def/geox) is a sub class of a general `Measure` class which is defined in a separate upper ontology. GeoSPARQL's upper classes are indeed the issue here: how far should GeoSPARQL go "up" in generalities?

> I agree that geometry is a special type of property of a FoI, but I would not call it a measurement

Why not? Isn't geometry a representation of a Feature's spatiality - it's particular value for a property from a set of all possible values, just as a Feature's  length is from all possible lengths or its temperature from all temperatures? What's so special about a Feature's spatial presence that it's not just another abstraction of a property of a thing?

> instead of directly discussing TBox, we can maybe start the discussion from data level (ABox)

ABox to illustrate potential instances, sure. But the ABox you are showing is from elsewhere and only represent that scenario's TBox in use, not what could be in use here.

> The (currently fictive) spatial property ex:length is not a sibling of omg:hasGeometry

Well perhaps you don't have it that way yet but the the position here is that it could be. Saying it currently isn't now is no argument for or against what it could be.

> I don't see much advantages except for more clarity on an abstract level

I am concerned about the orderliness of the representations of dimensions of things so I care about this issue of whether properties - spatial and others - can be aligned in their representation structure vis a vis a thing. I think that this is important for the complexity scaling of models (when things like temporality are added). If spatiality and other properties can be aligned they should be. If alignment causes problems, they shouldn't be. 

We can't know all possible future uses of these dimensions of things but we should try and aim for cohesion of representation, unless it proves problematic.

> Personally, I think that ABox links between the property and the geometry are more interesting, e.g. to indicate that the length property is measured on (derived from) the geometry object.

Sure, that _may_ be interesting but it doesn't aid or prevent the TBox modelling one way or the other. Measures could be derived from other measures, for example an average temperature of a Feature taken from a range of temperatures or, back to spatial land, a centroid from a polygon.

Reply:
Hi @nicholascar , thanks for all the pointers and elaborate explanations :). Noted that Geox is documented nicely with ABox examples.

I think I'm just a little distracted by the word "measurement", which I connect mentally to something that is measured physically, e.g. as used in the sentence "_I measure the width of the road using a total station_". In the construction domain a division is often made between such measured properties (as-is or as-built) vs designed properties (as-designed, meaning the road might not exist yet so there's nothing you can physically measure. As a designer you can declare that the width is equal to a certain value.). 

If the group decides to include `SpatialMeasurement` (I would go for `SpatialProperty` but maybe I'm the only one distracted by measurement) as a superclass for `Geometry`, is it then possible to add the above as a note/description to this superclass?

Reply:
Measure/measurement are rather fraught, can be both verbs and nouns :-( 
I snipped those diagrams from ISO 19103 primarily to clarify to the team that there is a class (datatype) `Measure` in our ecosystem, which is simply for scaled numbers, and the class `SpatialMeasure` in GEOX was modelled on that.

Reply:
(I know that was the intention because I created it in GEOX.)

Reply:
I'll resolve this comment to hide it in this Conversation but noted that there's an "If the group decides..." above so we might have to follow-up the output of this PR with another.